From ae9663d9bbf0523c9d77dc8bab7c10ed86f9d5f6 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Tue, 24 Jun 2008 16:56:49 +0800
Subject: [PATCH 025/449] ENGR00081573 Enhance MX35 unit test codes

Remove unsupported modules from Makefile. Modify fb and V4L2 test script.
Add Watchdog module.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 module_test/Makefile               |  1 +
 test-utils.sh                      |  2 ++
 test/demo/Makefile                 |  2 +-
 test/mxc_fb_test/autorun-fb.sh     |  8 +++++++-
 test/mxc_firi_test/Makefile        |  2 +-
 test/mxc_i2c_test/Makefile         |  4 ++--
 test/mxc_ipc_test/Makefile         |  4 ++--
 test/mxc_keyb_test/Makefile        |  4 ++--
 test/mxc_mpeg4_enc_test/Makefile   |  2 +-
 test/mxc_mu_test/Makefile          |  2 +-
 test/mxc_owire_test/Makefile       |  2 +-
 test/mxc_pm_test/Makefile          |  4 ++--
 test/mxc_pmic_test/Makefile        |  6 +++---
 test/mxc_rng_test/Makefile         |  4 ++--
 test/mxc_sahara_test/Makefile      |  2 +-
 test/mxc_scc/Makefile              |  2 +-
 test/mxc_security_test/Makefile    |  2 +-
 test/mxc_udma_test/Makefile        |  4 ++--
 test/mxc_v4l2_test/autorun-v4l2.sh | 10 +++++++++-
 19 files changed, 42 insertions(+), 25 deletions(-)

diff --git a/module_test/Makefile b/module_test/Makefile
index 1c58990..c63a4e7 100644
--- a/module_test/Makefile
+++ b/module_test/Makefile
@@ -13,6 +13,7 @@ module_list-$(CONFIG_MXC_SECURITY_RNG) += rng_test_driver.o
 module_list-$(CONFIG_MXC_SECURITY_HAC) += mxc_HAC_test.o
 module_list-$(CONFIG_MXC_SAHARA) += sahara_test_driver.o
 module_list-$(CONFIG_MXC_HMP4E) += memalloc.o
+module_list-$(CONFIG_WATCHDOG) += mxc_wdog_tm.o
 
 ifeq ($(CONFIG_MXC_SDMA_API),y)
 module_list-y += mxc_sdma_mem_test.o
diff --git a/test-utils.sh b/test-utils.sh
index 9eb0f5c..28dfbe5 100755
--- a/test-utils.sh
+++ b/test-utils.sh
@@ -168,6 +168,8 @@ platform()
 				* )	plat=IMX32ADS ;;
 			esac ;;
 
+		MX313StackBoard )	plat=IMX31_3STACK ;;
+		MX353StackBoard )	plat=IMX35_3STACK ;;
 		MX373StackBoard )	plat=IMX37_3STACK ;;
 
 		* )     ;;
diff --git a/test/demo/Makefile b/test/demo/Makefile
index a7af68f..b89cf2b 100644
--- a/test/demo/Makefile
+++ b/test/demo/Makefile
@@ -3,7 +3,7 @@
 #
 
 # list of platforms which did not want this test case
-EXCLUDE_LIST:= 
+EXCLUDE_LIST:= IMX35_3STACK
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 CONFIG_DEMO_DPTCHI=y
diff --git a/test/mxc_fb_test/autorun-fb.sh b/test/mxc_fb_test/autorun-fb.sh
index 9c914e6..1be6166 100755
--- a/test/mxc_fb_test/autorun-fb.sh
+++ b/test/mxc_fb_test/autorun-fb.sh
@@ -15,7 +15,6 @@ export TERM=linux
 # Turn off blanking
 setterm -blank 0 > /dev/tty0
 
-
 # Blank test
 echo FB Blank test
 echo -n 3 > /sys/class/graphics/fb0/blank
@@ -25,11 +24,16 @@ echo -n 0 > /sys/class/graphics/fb0/blank
 
 # Color tests
 echo FB Color test
+if [ "$(platform)" = IMX31_3STACK ] || [ "$(platform)" = IMX35_3STACK ]; then
+	bpp_list="16"
+else
 if [ "$(platform)" = MXC27530EVB ]; then
 	bpp_list="16 24"
 else
 	bpp_list="16 24 32"
 fi
+fi
+
 for bpp in $bpp_list;
 do
 	echo Setting FB to $bpp-bpp
@@ -57,6 +61,7 @@ done
 # Pan test
 #
 echo FB panning test
+if [ ! "$(platform)" = IMX35_3STACK ]; then
 echo 240,640 > /sys/class/graphics/fb0/virtual_size
 if ! grep -sq 240,640 /sys/class/graphics/fb0/virtual_size;
 then
@@ -68,6 +73,7 @@ then
 	echo FAIL - Unable to set virtual size
 	STATUS=1
 fi
+fi
 
 for i in $(seq 1 50); do
 	echo This is line $i. > /dev/tty0
diff --git a/test/mxc_firi_test/Makefile b/test/mxc_firi_test/Makefile
index bb42ee8..3ec8d2f 100644
--- a/test/mxc_firi_test/Makefile
+++ b/test/mxc_firi_test/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="MXC27520EVB MXC27530EVB IMX27ADS MXC30020EVB MXC30031ADS IMX37_3STACK"
+EXCLUDE_LIST:="MXC27520EVB MXC27530EVB IMX27ADS MXC30020EVB MXC30031ADS IMX37_3STACK IMX35_3STACK"
 
 # list of platforms which do not have either SIR or FIR
 IR_EXCLUDE_LIST:=
diff --git a/test/mxc_i2c_test/Makefile b/test/mxc_i2c_test/Makefile
index a7ddda4..fae1f15 100644
--- a/test/mxc_i2c_test/Makefile
+++ b/test/mxc_i2c_test/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:= 
+EXCLUDE_LIST:="IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = \
@@ -8,7 +8,7 @@ OBJS = \
 	$(OBJDIR)/mxc_i2c_test_samsung_cam.out \
 	$(OBJDIR)/mxc_i2c_test_magna_cam.out
 else
-OBJS = 
+OBJS =
 endif
 
 all : $(OBJS)
diff --git a/test/mxc_ipc_test/Makefile b/test/mxc_ipc_test/Makefile
index 644611f..262a4dc 100644
--- a/test/mxc_ipc_test/Makefile
+++ b/test/mxc_ipc_test/Makefile
@@ -1,10 +1,10 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="IMX21ADS IMX27ADS IMX31ADS MXC91131EVB IMX37_3STACK"
+EXCLUDE_LIST:="IMX21ADS IMX27ADS IMX31ADS MXC91131EVB IMX37_3STACK IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = $(OBJDIR)/mxc_ipc_test.out
 else
-OBJS = 
+OBJS =
 endif
 
 all : $(OBJS)
diff --git a/test/mxc_keyb_test/Makefile b/test/mxc_keyb_test/Makefile
index f0358ec..83c95eb 100644
--- a/test/mxc_keyb_test/Makefile
+++ b/test/mxc_keyb_test/Makefile
@@ -1,10 +1,10 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:= 
+EXCLUDE_LIST:="IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = $(OBJDIR)/autorun-keypad.sh $(OBJDIR)/mxc_keyb_test.sh
 else
-OBJS = 
+OBJS =
 endif
 
 all : $(OBJS)
diff --git a/test/mxc_mpeg4_enc_test/Makefile b/test/mxc_mpeg4_enc_test/Makefile
index 5fc38b7..e7e2999 100644
--- a/test/mxc_mpeg4_enc_test/Makefile
+++ b/test/mxc_mpeg4_enc_test/Makefile
@@ -5,7 +5,7 @@ OBJDIR=$(TOPDIR)/platform/$(PLATFORM)/$(MODULE_NAME)
 CFLAGS += -I . -lpthread
 
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="IMX21ADS IMX27ADS IMX37_3STACK"
+EXCLUDE_LIST:="IMX21ADS IMX27ADS IMX37_3STACK IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 ifeq "$(PLATFORM)" "MXC30030EVB"
diff --git a/test/mxc_mu_test/Makefile b/test/mxc_mu_test/Makefile
index 9f8acae..2bec6f6 100644
--- a/test/mxc_mu_test/Makefile
+++ b/test/mxc_mu_test/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="IMX27ADS IMX31ADS IMX37_3STACK"
+EXCLUDE_LIST:="IMX27ADS IMX31ADS IMX37_3STACK IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = \
diff --git a/test/mxc_owire_test/Makefile b/test/mxc_owire_test/Makefile
index 802ee65..88bf29e 100644
--- a/test/mxc_owire_test/Makefile
+++ b/test/mxc_owire_test/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="IMX31_3STACK IMX37_3STACK"
+EXCLUDE_LIST:="IMX31_3STACK IMX37_3STACK IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = $(OBJDIR)/autorun-owire.sh
diff --git a/test/mxc_pm_test/Makefile b/test/mxc_pm_test/Makefile
index c93abff..a913723 100644
--- a/test/mxc_pm_test/Makefile
+++ b/test/mxc_pm_test/Makefile
@@ -1,10 +1,10 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:= 
+EXCLUDE_LIST:="IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = $(OBJDIR)/mxc_pm_test.out
 else
-OBJS = 
+OBJS =
 endif
 
 all : $(OBJS)
diff --git a/test/mxc_pmic_test/Makefile b/test/mxc_pmic_test/Makefile
index 0a4b069..c4988b1 100644
--- a/test/mxc_pmic_test/Makefile
+++ b/test/mxc_pmic_test/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:= IMX37_3STACK
+EXCLUDE_LIST:="IMX37_3STACK IMX35_3STACK"
 
 LTP_DIR = ../
 LOADLIBES+= -L$(LTP_DIR)lib -lltp -lpthread -lm
@@ -7,7 +7,7 @@ LOADLIBES+= -L$(LTP_DIR)lib -lltp -lpthread -lm
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = libltp.a $(OBJDIR)/autorun-pmic.sh
 else
-OBJS = 
+OBJS =
 endif
 
 CFLAGS+= -Wall -I../include
@@ -26,7 +26,7 @@ SUBDIRS =       protocol_tests \
 		pmic_testapp_light \
 		pmic_testapp_battery \
 		pmic_testapp_power \
-		pmic_testapp_rtc 
+		pmic_testapp_rtc
 
 all: $(OBJS)
 
diff --git a/test/mxc_rng_test/Makefile b/test/mxc_rng_test/Makefile
index c8433a7..c21bab4 100644
--- a/test/mxc_rng_test/Makefile
+++ b/test/mxc_rng_test/Makefile
@@ -5,7 +5,7 @@
 #****************************************************************************/
 
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="IMX27ADS IMX27EVB MXC27530EVB MXC30030ADS MXC30030EVB MXC30031ADS"
+EXCLUDE_LIST:="IMX27ADS IMX27EVB MXC27530EVB MXC30030ADS MXC30030EVB MXC30031ADS IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = apitest rng_test move
@@ -22,7 +22,7 @@ endif
 ifeq ("$(PLATFORM)", "IMX31ADS")
 TARGET_ARCH=-DCONFIG_ARCH_MX3
 endif
-ifeq ("$(PLATFORM)", "IMX31_3DS")
+ifeq ("$(PLATFORM)", "IMX31_3STACK")
 TARGET_ARCH=-DCONFIG_ARCH_MX3
 endif
 ifeq ("$(PLATFORM)", "MXC91131EVB")
diff --git a/test/mxc_sahara_test/Makefile b/test/mxc_sahara_test/Makefile
index 08af427..a55dff8 100644
--- a/test/mxc_sahara_test/Makefile
+++ b/test/mxc_sahara_test/Makefile
@@ -3,7 +3,7 @@
 ###################################################################
 
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="MXC91131EVB I30030ADS I30030EVB IMX21ADS IMX31ADS IMX31_3DS MXC30020EVB MXC27520EVB"
+EXCLUDE_LIST:= MXC91131EVB I30030ADS I30030EVB IMX21ADS IMX31ADS IMX31_3STACK MXC30020EVB MXC27520EVB IMX35_3STACK
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = apitest
diff --git a/test/mxc_scc/Makefile b/test/mxc_scc/Makefile
index ac25442..00cd641 100644
--- a/test/mxc_scc/Makefile
+++ b/test/mxc_scc/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:="IMX21ADS MXC30031ADS IMX31_3DS"
+EXCLUDE_LIST:="IMX21ADS MXC30031ADS IMX31_3STACK IMX35_3STACK"
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = $(OBJDIR)/scc_test.out $(OBJDIR)/autorun-scc.sh
diff --git a/test/mxc_security_test/Makefile b/test/mxc_security_test/Makefile
index 24e27eb..3e50f22 100644
--- a/test/mxc_security_test/Makefile
+++ b/test/mxc_security_test/Makefile
@@ -1,5 +1,5 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:=IMX37_3STACK
+EXCLUDE_LIST:="IMX37_3STACK IMX35_3STACK"
 
 # list of platforms that have hacc
 HACC_LIST:=MXC91131EVB
diff --git a/test/mxc_udma_test/Makefile b/test/mxc_udma_test/Makefile
index 67126a4..b07886c 100644
--- a/test/mxc_udma_test/Makefile
+++ b/test/mxc_udma_test/Makefile
@@ -1,10 +1,10 @@
 # list of platforms which did not want this test case
-EXCLUDE_LIST:= IMX31ADS I30030EVB I30030ADS MXC30030EVB MXC30030ADS MXC91131EVB MXC27530EVB
+EXCLUDE_LIST:= IMX31ADS I30030EVB I30030ADS MXC30030EVB MXC30030ADS MXC91131EVB MXC27530EVB IMX35_3STACK
 
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = $(OBJDIR)/mxc_udma_test.out
 else
-OBJS = 
+OBJS =
 endif
 
 all : $(OBJS)
diff --git a/test/mxc_v4l2_test/autorun-v4l2.sh b/test/mxc_v4l2_test/autorun-v4l2.sh
index 9b31fba..d9d20b6 100755
--- a/test/mxc_v4l2_test/autorun-v4l2.sh
+++ b/test/mxc_v4l2_test/autorun-v4l2.sh
@@ -15,12 +15,20 @@ fi
 
 # devnode test
 if [ $TEST_CAMERA = 1 ]; then
+if [ "$(platform)" = "IMX35_3STACK" ]; then
+insmod ipu_prp_enc
+insmod ipu_prp_vf_sdc
+insmod ipu_prp_vf_sdc_bg
+insmod ipu_still
+insmod ov2640_camera
+insmod mxc_v4l2_capture
+fi
 check_devnode "/dev/video0"
 fi
 check_devnode "/dev/video16"
 
 # Turn off fb blanking
-echo -e "\033[9;0]" > /dev/vc/0
+echo -e "\033[9;0]" > /dev/tty0
 
 if [ "$(platform)" = IMX27ADS ]; then
 DISPLAY=0;
-- 
1.8.0

